pkg/health/health_aws.go

```diff
@@ -12,94 +15,105 @@ const (
 	AWSResourceTypeSubnet string = "subnet"
 )
 
-// MapAWSStatus maps an AWS resource's statuses to a Health Code
-func MapAWSStatus(status string, resourceType string) string {
-	if resourceStatuses, found := awsStatusMap[resourceType]; found {
+func GetAWSResourceHealth(resourceType, status string, deletionTimeStamp *time.Time) (health HealthStatus) {
+	if deletionTimeStamp != nil {
+		return HealthStatus{
+			Status: HealthStatusDeleting,
+			Health: HealthUnknown,
+		}
+	}
+
+	if resourceStatuses, found := awsResourceHealthmap[resourceType]; found {
 		if v, found := resourceStatuses[strings.ToLower(status)]; found {
-			return string(v)
+			return v
 		}
 	}
 
-	return string(HealthStatusUnknown)
+	return HealthStatus{
+		Status: HealthStatusUnknown,
+		Health: HealthUnknown,
+		Ready:  false,
+	}
 }
 
-var awsStatusMap = map[string]map[string]HealthStatusCode{
+var awsResourceHealthmap = map[string]map[string]HealthStatus{
 	AWSResourceTypeEC2: {
-		"pending":       HealthStatusPending,
-		"running":       HealthStatusHealthy,
-		"shutting-down": HealthStatusDeleting,
-		"stopped":       HealthStatusStopped,
-		"stopping":      HealthStatusStopping,
-		"terminated":    HealthStatusDeleted,
+		"pending":       HealthStatus{Status: HealthStatusPending, Health: HealthUnknown},
+		"running":       HealthStatus{Status: HealthStatusHealthy, Health: HealthHealthy, Ready: true},
+		"shutting-down": HealthStatus{Status: HealthStatusDeleting, Health: HealthUnhealthy},
```

Comment:
An instance that is stopped or stopping has a health of unknown - only if it failed to start would be unhealthy